docs/messaging/kafka.md

```diff
@@ -44,7 +44,63 @@ If an intermediary broker is present, `service.name` and `peer.service` will not
 
 `messaging.client_id` SHOULD be set to the `client-id` of consumers, or to the `client.id` property of producers.
 
-## Examples
+## Metrics
+
+### Metric: `messaging.publish.duration`
+
+Measures time it takes to publish message or a batch of messages to Apache Kafka including all retries.
+
+This metric is [required][MetricRequired]
+
+This metric follows the common [messaging.publish.duration](./messaging-metrics.md#metric-messagingpublishduration) definition.
+
+**Notes:**
+
+- In addition to generic attributes defined for the metric, Apache Kafka instrumentation SHOULD report following attributes:
+  * `messaging.kafka.destination.partition`
+
+### Metric: `messaging.publish.messages`
+
+Measures the number of published messages.
+
+This metric is [required][MetricRequired]
+
+This metric follows the common [messaging.publish.messages](./messaging-metrics.md#metric-messagingpublishmessages) definition.
+
+**Notes:**
+
+- In addition to generic attributes defined for the metric, Apache Kafka instrumentation SHOULD report following attributes:
+  * `messaging.kafka.destination.partition`
+
+### Metric: `messaging.receive.duration`
+
+Measures duration of consumer client call that polls messages.
+
+This metric is [required][MetricRequired].
+
+This metric follows the common [messaging.receive.duration](./messaging-metrics.md#metric-messagingdeliverduration) definition.
+
+**Notes:**
+
+- In addition to generic attributes defined for the metric, Apache Kafka instrumentation SHOULD report following attributes:
+  * `messaging.kafka.destination.partition`
+  * `messaging.kafka.consumer.group`
+
+### Metric: `messaging.receive.messages`
+
+Measures the number of received messages.
+
+This metric is [required][MetricRequired].
+
+This metric follows the common [messaging.receive.messages](./messaging-metrics.md#metric-messagingreceivemessages) definition.
+
+**Notes:**
+
+- In addition to generic attributes defined for the metric, Apache Kafka instrumentation SHOULD report following attributes:
```

Comment:
> Apache Kafka instrumentation SHOULD report following attributes:

Instead of the SHOULD requirement, it would be better to define system-specific attributes consistently with attribute requirement levels ("Recommended", "Required", or "Conditionally required").

Reply:
Good point. Without SHOULD it does not sound right "Kafka instrumentations report ..."

this problem will disappear if we could extend generic metric in yaml and render attributes,
In the mean time I changed to 
"<System> instrumentations SHOULD report the following attributes (according to their requirement levels)" with link to the table with attributes.

LMK what you think